internal/controller/management/projects/projects.go

```diff
@@ -480,23 +480,30 @@ func (r *reconciler) ensureNamespace(ctx context.Context, project *kargoapi.Proj
 		// uncommon scenario where an organization has its own controller that
 		// creates and initializes namespaces to ensure compliance with
 		// internal policies. Such a controller might already own the namespace.
-		updated, err := r.ensureFinalizerFn(ctx, r.client, ns)
-		if err != nil {
-			return fmt.Errorf("error ensuring finalizer on namespace %q: %w", project.Name, err)
-		}
-		if updated {
-			logger.Debug("added finalizer to namespace")
-		}
-		ns.OwnerReferences = append(ns.OwnerReferences, *ownerRef)
-		if err = r.patchOwnerReferencesFn(ctx, r.client, ns); err != nil {
-			return fmt.Errorf(
-				"error patching namespace %q with project %q as owner: %w",
-				project.Name,
-				project.Name,
-				err,
-			)
+
+		// Before setting project as the owner for the existing namespace check for `keep-namespace`
+		// label in either project/namespace if it's true we don't want to make project as the owner
+		if project.Labels[kargoapi.KeepNamespaceLabelKey] != kargoapi.LabelTrueValue &&
+			ns.Labels[kargoapi.KeepNamespaceLabelKey] != kargoapi.LabelTrueValue {
+
+			updated, err := r.ensureFinalizerFn(ctx, r.client, ns)
+			if err != nil {
+				return fmt.Errorf("error ensuring finalizer on namespace %q: %w", project.Name, err)
+			}
+			if updated {
+				logger.Debug("added finalizer to namespace")
+			}
```

Comment:
I'm not sure this should be conditional. This is for ensuring things work in the _opposite_ direction. i.e. Ensures that if you delete the Project's namespace, the Project goes with it. I don't feel as if that is behavior that needs to change in order to prevent the namespace from being deleted when the Project is.